spec/acceptance/autoscaling_spec.rb

```diff
@@ -0,0 +1,74 @@
+require 'spec_helper_acceptance'
+require 'securerandom'
+
+describe "ec2_autoscalinggroup" do
+
+  before(:all) do
+    @default_region = 'sa-east-1'
+    @aws = AwsHelper.new(@default_region)
+  end
+
+  def find_autoscaling_group(name)
+    groups = @aws.get_autoscaling_groups(name)
+    expect(groups.count).to eq(1)
+    groups.first
+  end
+
+  def find_launch_config(name)
+    config = @aws.get_launch_configs(name)
+    expect(config.count).to eq(1)
+    config.first
+  end
+
+  def find_scaling_policy(name, group)
+    policy = @aws.get_scaling_policies(name, group)
+    expect(policy.count).to eq(1)
+    policy.first
+  end
+
+  def find_alarm(name)
+    alarm = @aws.get_alarms(name)
+    expect(alarm.count).to eq(1)
+    alarm.first
+  end
+
+  describe 'should create a new autoscaling group' do
+
+    before(:all) do
+      @name = "#{PuppetManifest.env_id}-#{SecureRandom.uuid}"
+      @config = {
+        :name => @name,
+        :ensure => 'present',
+      }
+
+      template = 'autoscaling.pp.tmpl'
+      PuppetManifest.new(template, @config).apply
+    end
+
+    after(:all) do
+      new_config = @config.update({:ensure => 'absent'})
+      template = 'autoscaling_delete.pp.tmpl'
+      PuppetManifest.new(template, new_config).apply
+    end
+
+    it 'should create an auto scaling group' do
+      find_autoscaling_group("#{@name}-asg")
+    end
+
+    it 'should create a launch configuration' do
+      find_launch_config("#{@name}-lc")
+    end
+
+    it 'should create CloudWatch alarms' do
+      find_alarm("#{@name}-AddCapacity")
+      find_alarm("#{@name}-RemoveCapacity")
+    end
+
+    it 'should create scaling policies' do
+      find_scaling_policy("#{@name}-scaleout", "#{@name}-asg")
+      find_scaling_policy("#{@name}-scalein", "#{@name}-asg")
+    end
+
```

Comment:
I've added a bunch more tests here to assert on various properties, and to change those properties.

I've not added deletion tests as the waiters aren't implemented for the autoscaling resources yet.  We can add our own implementation if we thing it's worth it, but I'd probably rather wait and add them later with the native wait.